### ISSUE TITLE
[FEATURE] Heating Boost on_off

### ISSUE THREAD

**Issue by johnsdixon** [Pyhass/Hive-Custom-Component#59]
**Is your feature request related to a problem? Please describe.**
I like that you can enable and disable boost for the hot water heating. I use this to add top-up heating to my hot water when the heating is on, saving energy later.
I am frustrated that we can only boost heating into the ON state, and there is no option to disable the boost once it has started, except to wait for the duration to expire, or using the Hive app.

**Describe the solution you'd like**
I would like the ability to use the hive.boost_heating service by sending data with the `on_off: "off"` option (as with the hot water service) to disable a boost started via HA or the Hive app.

**Describe alternatives you've considered**
Other options may be a new service (hive.stop_boost), but this would mean more integration work and a new service handler.

**Additional context**
N/A

**Comment by lwsrbrts** [comment]
As a workaround, I have been able to successfully set a 1 minute boost on the heating - this should work to get it to turn off sooner than the time originally set, but I agree that have a cancel_boost service would be more logical.

```
entity_id: climate.heating
time_period: '00:01:00'
temperature: 20.5
```

Similarly you could set a lower temperature and for 1 minute to get it to turn off immediately but with that 1 minute pause before the boost "state" disappears.

I spent a good amount of time reversing the Hive API for PowerShell and the implementation of the "Cancel" boost mode relies on knowing the previous state of the heating prior to boosting such that the cancel actually sets the heating mode back to that - but this is exposed in the return data for the heating appliance, so easy enough.

My logic for setting the heating mode to its previous state starts here in my own repo:

https://github.com/lwsrbrts/PoSHive/blob/8288c21891cfbaf51e03c432c58b0189f4df9509/PoSHive.ps1#L589

Worth mentioning that I'm more than happy for my project to be used, even if just to save time finding a property name. At this stage, with Hive's change to the login method and SSO, I'm struggling to get it working anyway.

**Comment by johnsdixon** [comment]
@lwsrbrts, thanks for the workaround. Not quite as neat as a 'proper' cancel function, but achieves nearlythe same result, apart from that delay.

**Comment by KJonline** [comment]
Please keep this one open I will build a service for this, we already have all the data. Once I have completed the current PR for home assistant I can look into these changes.

**Comment by KJonline** [comment]
This is included in home assistant 2021.5.0
